File: Projects/rosalind/PMCH.java
import java.io.*;
import java.util.*;
import java.math.BigInteger;

public class PMCH {
	private static final Integer A = (int) 'A', G = (int) 'G';
	private static BigInteger factorial(long i) {
		return i == 0 ? BigInteger.ONE : factorial(i-1).multiply(BigInteger.valueOf(i));
	}
	public static void main(String args[]) throws IOException {
		Scanner in = new Scanner(System.in);
		in.next();
		String seq = "";
		while (in.hasNext())
			seq += in.next();
		System.out.println(factorial(seq.chars().filter(A::equals).count()).multiply(factorial(seq.chars().filter(G::equals).count())));
		in.close();
	}
}

File: Projects/rosalind/PMCH.kt
package rosalind

import java.io.IOException
import java.math.BigInteger
import java.util.*

object PMCH {
    private const val A = 'A'.toInt()
    private const val G = 'G'.toInt()
    private fun factorial(i: Long): BigInteger {
        return if (i == 0L) BigInteger.ONE else factorial(i - 1).multiply(BigInteger.valueOf(i))
    }

    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        stdin.next()
        var seq = ""
        while (stdin.hasNext()) seq += stdin.next()
        println(factorial(seq.chars().filter { obj: Int -> A.equals(obj) }.count()).multiply(factorial(seq.chars().filter { obj: Int -> G.equals(obj) }.count())))
        stdin.close()
    }
}